go/vt/tabletserver/tx_pool.go

```diff
@@ -141,6 +141,22 @@ func (axp *TxPool) WaitForEmpty() {
 // Begin begins a transaction, and returns the associated transaction id.
 // Subsequent statements can access the connection through the transaction id.
 func (axp *TxPool) Begin(ctx context.Context) (int64, error) {
+	return axp.internalBegin(ctx, func() int64 {
```

Comment:
TxPool need not change. Instead, change add BeginAgain to TxConn. Just execute a `commit` and `begin`. The commit ensures that anything accidentally uncommitted will not get lost.